Mainnet/STOContract/STOContract/STOContract.cs

```diff
@@ -0,0 +1,1361 @@
+    {
+        Assert(tokenType < 3, $"The {nameof(tokenType)} parameter can be between 0 and 2");
+
+        Assert(PersistentState.IsContract(kycAddress), "The kycAdress is not a contract adress");
+
+        var periods = Serializer.ToArray<SalePeriodInput>(salePeriods);
+
+        ValidatePeriods(periods);
+        var tokenTypeEnum = (TokenType)tokenType;
+        var result = CreateTokenContract(tokenTypeEnum, totalSupply, name, symbol);
+
+        Assert(result.Success, "Creating token contract failed.");
+
+        Log(new STOSetupLog { TokenAddress = result.NewContractAddress });
+
+        KYCAddress = kycAddress;
+        MapperAddress = mapperAddress;
+        TokenAddress = result.NewContractAddress;
+        IsNonFungibleToken = tokenTypeEnum == TokenType.NonFungibleToken;
+        TokenBalance = IsNonFungibleToken ? ulong.MaxValue : totalSupply;
+        Owner = owner;
+        SetPeriods(periods);
+    }
+
+    private ICreateResult CreateTokenContract(TokenType tokenType, ulong totalSupply, string name, string symbol)
+    {
+        switch (tokenType)
+        {
+            case TokenType.StandardToken: return Create<StandardToken>(parameters: new object[] { totalSupply, name, symbol });
+            case TokenType.DividendToken: return Create<DividendToken>(parameters: new object[] { totalSupply, name, symbol });
+            default: return Create<NonFungibleToken>(parameters: new object[] { name, symbol });
+        }
+    }
+    public override void Receive() => Invest();
+
+    public bool Invest()
+    {
+        Assert(SaleOpen, "The STO is completed.");
+        Assert(Message.Value > 0, "The amount should be higher than zero");
+
+        EnsureKycVerified();
+
+        var saleInfo = GetSaleInfo();
+
+        var method = IsNonFungibleToken ? nameof(NonFungibleToken.MintAll) : nameof(IStandardToken.TransferTo);
+        var result = Call(TokenAddress, 0, method, new object[] { Message.Sender, saleInfo.TokenAmount });
+
+        Assert(result.Success && (bool)result.ReturnValue, "Token transfer failed.");
+
+        Log(new InvestLog { Sender = Message.Sender, Invested = saleInfo.Invested, TokenAmount = saleInfo.TokenAmount, Refunded = saleInfo.RefundAmount });
+
+        TokenBalance = checked(TokenBalance - saleInfo.TokenAmount);
+
+        if (saleInfo.RefundAmount > 0) // refund over sold amount
+            Transfer(Message.Sender, saleInfo.RefundAmount);
+
+        return true;
+    }
+
+    private void EnsureKycVerified()
+    {
+        var result = Call(MapperAddress, 0, "GetSecondaryAddress", new object[] { Message.Sender });
+
+        if (result.Success && result.ReturnValue is Address identityAddress && identityAddress != Address.Zero)
+        {
+            result = Call(KYCAddress, 0, "GetClaim", new object[] { identityAddress, (uint)3 /*shufti kyc*/ });
+
+            Assert(result.Success && result.ReturnValue != null, "Your KYC is not verified.");
```

Comment:
Will ReturnValue ever be null if `GetClaim` returns `byte[]`? For safety, might be worth checking there is a length on the result instead of checking not null.

Reply:
Updated

Reply:
Was this updated? I still see the same `Assert(result.Success && result.ReturnValue != null, "Your KYC is not verified.");`